website_sale_require_legal/controllers/main.py

```diff
@@ -34,3 +34,21 @@ def checkout_form_validate(self, mode, all_form_values, data):
         # Unpatch context
         request.context = old_context
         return result
+
+    def _checkout_form_save(self, mode, checkout, all_values):
+        res = super(WebsiteSale, self)._checkout_form_save(
+            mode, checkout, all_values)
+        if (all_values.get('submitted') and
+                all_values.get('accepted_legal_terms')):
+            environ = request.httprequest.headers.environ
+            metadata = "Website legal terms acceptance metadata: "
+            metadata += "\n".join(
+                "%s: %s" % (val, environ.get(val)) for val in (
+                        "REMOTE_ADDR",
+                        "HTTP_USER_AGENT",
+                        "HTTP_ACCEPT_LANGUAGE",
+                    )
+                )
+            partner_id = request.env['res.partner'].browse(res)
+            partner_id.message_post(body=metadata)
```

Comment:
I imagine this message should be of type log. Don't you need to specify that? We don't want this to reach the buyer, right?